source/classes.tex

```diff
@@ -1873,11 +1873,13 @@
 a base class copy/move assignment operator is always hidden
 by the corresponding assignment operator of a derived class\iref{over.ass}.
 \begin{note}
-A \grammarterm{using-declaration} that names an assignment operator
-with a parameter type that could be that of a
-copy/move assignment operator for a
-derived class does not suppress the implicit declaration of the derived class
-operator\iref{namespace.udecl}.
+A \grammarterm{using-declaration}\iref{namespace.udecl}
+that brings in an assignment operator from a base class
+never suppresses the implicit declaration of
+an assignment operator of the derived class,
+even if the base class assignment operator,
+when considered as a member of the derived class,
+is a copy assignment operator or a move assignment operator.
```

Comment:
I don't like the "is" because the assignment operator never is one of those things.  I already provided my own [variation](https://github.com/cplusplus/draft/issues/4388#issuecomment-735898621) that avoids that phrasing, but maybe it's inferior in some other way.

Reply:
Yes, I see what you mean, and I agree. Something like "would" is really what we'd need here.

Reply:
Merging the two suggestions, how about:

> A using-declaration in a derived class `C` that brings in an assignment operator from a base class never suppresses the implicit declaration of an assignment operator of `C`, even if the base class assignment operator would be a copy or move assignment operator if declared as a member of `C`.

Reply:
Looks fine, except that "brings in" is a reversion of P1787R6's "names", which says that *using-declaration*s do not copy-paste declarations around.  (Admittedly, this is (now) a note, but I don't want to encourage the old way of thinking.)

Reply:
Ah, I see. How about this then:

> A using-declaration in a derived class `C` that names an assignment operator from a base class never suppresses the implicit declaration of an assignment operator of `C`, even if the base class assignment operator would be a copy or move assignment operator if declared as a member of `C`.

Reply:
You can't name an assignment operator other than from a base class, but maybe it helps readability to explain it in more words like this.  In any case, ship it.

Reply:
You mean that "from a base class" is redundant? If so, then yes, that's true, but in a note this seems helpful explanation, since it's exactly the puzzle solving that would be needed in the absence of this explanation that the note tries to spell out. (We could also say "an assignment operator (from a base class)", or "an assignment operator, which must be from a base class". But we shan't.)

Reply:
Fixed